app/code/Magento/Developer/Model/Di/PluginList.php

```diff
@@ -143,9 +143,8 @@ private function addPluginToList($pluginInstance, $pluginMethod, $methodTypes, $
     {
         if ($methodTypes & $typeCode) {
             if (!array_key_exists($pluginInstance, $this->pluginList[$this->pluginTypeMapping[$typeCode]])) {
-                $this->pluginList[$this->pluginTypeMapping[$typeCode]][$pluginInstance] = [];
+                $this->pluginList[$this->pluginTypeMapping[$typeCode]][$pluginInstance][] = $pluginMethod;
             }
-            $this->pluginList[$this->pluginTypeMapping[$typeCode]][$pluginInstance][] = $pluginMethod ;
```

Comment:
```
$this->pluginList[$this->pluginTypeMapping[$typeCode]][$pluginInstance][] = $pluginMethod ; ->

$this->pluginList[$this->pluginTypeMapping[$typeCode]][$pluginInstance][$pluginMethod] = 1
```

Please avoid duplication by reversing array. Please check if it is possible to remove `if` while avoiding PHP notice.